### PR TITLE
Watch and warn on rename or delete events for the root index.html file

### PR DESCRIPTION
**Pull request by humphd**
This is part of https://github.com/mozilla/thimble.mozilla.org/issues/1569#issuecomment-258889724.  It adds an extra extension (not loaded by default--it will need to be enabled in Thimble) to watch and warn on any delete or rename to the project's root `index.html` file.

To test this, do the following:
* `npm run build`
* `npm start`
* load http://localhost:8000/src/hosted.html?enableExtensions=bramble-watch-index.html (NOTE: the extension must be manually enabled in the URL)
* Make sure you have an `index.html` file, then try right-clicking and `Rename` and `Delete`.  Both should show a modal dialog with info.

I'm not sure if the text I've got in the dialog is right.  I want to do another patch in Thimble that offers to create an `index.html` on publish, but decided to only warn about the issue here.

I've also added jshint excludes for `src/nls` to stop jshint complaining about strings as if they are code, and breaking the build.

r? @gideonthomas, @flukeout

### CODE REVIEW

File: locales/en-US/editor.properties [hunk 1]
```diff
@@ -189,3 +189,8 @@ ERROR_MOVING_FILE_DIALOG_HEADER=Move Error
 UNEXPECTED_ERROR_MOVING_FILE=An unexpected error occurred when attempting to move {0} to {1}
 # {0} is the name of the file/folder being moved and {1} is the name of the folder it is being moved to
 ERROR_MOVING_FILE_SAME_NAME=A file or folder with the name {0} already exists in {1}. Consider renaming either one to continue.
+
+# extensions/extra/bramble-watch-index.html
+
+INDEX_HTML_FILE_MISSING_DIALOG_TITLE=Warning - Missing index.html file
```
Comment (gideonthomas) [inline locales/en-US/editor.properties]: Before this lands, we should remove these changes from the commit and add them into Thimble so that it will show up on Pontoon to be localized and when Brackets syncs this file, there won't be merge conflicts.

File: locales/en-US/editor.properties [hunk 2]
```diff
@@ -189,3 +189,8 @@ ERROR_MOVING_FILE_DIALOG_HEADER=Move Error
...
+INDEX_HTML_FILE_MISSING_DIALOG_MESSAGE=Your project has no index.html file. Web servers look for index.html when URLs omit the filename, for example: google.com/ actually refers to google.com/index.html. Consider adding an index.html.
```
Comment (gideonthomas) [inline locales/en-US/editor.properties]: We also should simplify the wording for this a bit since when I was trying to explain this during a session, a lot of them were confused by "web servers".

Comment (humphd) [inline locales/en-US/editor.properties]: Ah, OK, I misunderstood how to do this properly.  The docs made me think I was supposed to do it here in Bramble.  I'll do a PR against Thimble to add them there and remove these once you r+.

File: locales/en-US/editor.properties (hunk 1 above)
Comment (humphd) [inline locales/en-US/editor.properties]: Ah, OK, I misunderstood how to do this properly.  The docs made me think I was supposed to do it here in Bramble.  I'll do a PR against Thimble to add them there and remove these once you r+.

**Comment by flukeout** [comment]
If you get a chance, can you copy and paste a screenshot of how this dialog looks? Don't have my env up and running at the moment. I'll work on some copy ideas and paste them here in a bit.

**Comment by flukeout** [comment]
I'm thinking of something along the lines of...

**You need an index.html page!**

Projects that don't contain an index.html page will appear broken to viewers, it's the first page that the browser tries to load.

**Question**: when does this dialog pop-up? As soon as you right click and delete / rename? Are there an ``Undo`` or ``Do it anyway`` type buttons on the dialog?

**Comment by humphd** [comment]
@gideonthomas I've updated this PR, and also split out the Thimble bits into https://github.com/mozilla/thimble.mozilla.org/pull/1580

**Comment by gideonthomas** [comment]
Ok looks like this works!

@humphd could you also add these three lines above this one: https://github.com/mozilla/brackets/blob/master/scripts/properties2js.js#L174
```js
var DSStoreIndex = folders.indexOf(".DS_Store");
if(DSStoreIndex > -1) {
    folders.splice(folders.indexOf(".DS_Store"), 1);
}
```
I was constantly blocked from proper l10n running on Macs.

After that I think this code is good to merge

**Comment by gideonthomas** [comment]
@flukeout here's the screenshot:
screen shot 2016-11-08 at 12 42 02 pm

**Comment by humphd** [comment]
I'm confused about strings again.  Are you saying I need the strings here (Bramble) *and* in Thimble? Or just in Thimble?

Also, I updated the strings to a version of what Luke wrote already.

**Comment by gideonthomas** [comment]
@humphd no sorry, the strings are fine in Thimble, you don't need them in Bramble. 

The code that I provide above (unrelated to your patch) just ignores `.DS_Store` when it runs `npm run localize` because it was throwing errors since it tries to localize files in `.DS_Store` on Macs.

**Comment by humphd** [comment]
@gideonthomas can you just take one last look at this?  I made your change, but also remembered that I had to manually add the new extension to the service worker precache list.

Also, remind me how you want this landed.  Do I squash, then land the PR in GitHub?  I can't remember your preference.

**Comment by gideonthomas** [comment]
r=me

**Comment by humphd** [comment]
@gideonthomas reasking..."Also, remind me how you want this landed. Do I squash, then land the PR in GitHub? I can't remember your preference."

**Comment by gideonthomas** [comment]
@humphd you can use the "squash and merge" button in this pr to merge it in.
